wfe/web-front-end.go

```diff
@@ -335,6 +329,7 @@ func (wfe *WebFrontEndImpl) NewAuthorization(response http.ResponseWriter, reque
 	authzURL := wfe.AuthzBase + string(authz.ID)
 	authz.ID = ""
 	authz.RegistrationID = 0
```

Comment:
The clearing above looks like it wouldn't be needed with a `PrepareForPublicMarshal` method